nautobot_netbox_importer/diffsync/netbox.py

```diff
@@ -0,0 +1,497 @@
+
+        if field.nautobot_name == "parent_id" and parent:
+            result = wrapper.get_pk_from_uid(parent)
+            wrapper.add_reference(result, wrapper)
+        elif location:
+            result = location_wrapper.get_pk_from_uid(location)
+            location_wrapper.add_reference(result, wrapper)
+        elif site:
+            result = site_wrapper.get_pk_from_uid(site)
+            site_wrapper.add_reference(result, wrapper)
+        elif region:
+            result = region_wrapper.get_pk_from_uid(region)
+            region_wrapper.add_reference(result, wrapper)
+        else:
+            result = None
+
+        target[field.nautobot_name] = result
+
+    field.set_importer(importer)
+
+
+def _define_units(field: SourceField) -> None:
+    field.set_nautobot_field(field.name)
+
+    def importer(source: RecordData, target: RecordData) -> None:
+        # NetBox 3.4 units is `list[int]`, previous versions are JSON string with list of strings
+        units = source.get(field.name, None)
+        if units in EMPTY_VALUES:
+            return
+
+        if isinstance(units, str):
+            units = json.loads(units)
+            if units:
+                units = [int(unit) for unit in units]
+
+        target[field.name] = units
+
+    field.set_importer(importer)
+
+
+def _role_definition_factory(adapter: SourceAdapter, role_content_type: str):
+    if role_content_type in adapter.wrappers:
+        role_wrapper = adapter.wrappers[role_content_type]
+    else:
+        role_wrapper = adapter.configure_model(
+            role_content_type,
+            nautobot_content_type="extras.role",
+            fields={"color": "color", "content_types": "content_types"},
+        )
+
+    def definition(field: SourceField) -> None:
+        field.set_role_importer(role_wrapper)
+
+    return definition
+
+
+def _setup_source() -> SourceAdapter:
+    """Setup NetBox models importers."""
+    adapter = SourceAdapter(name="NetBox")
+    adapter.configure(
+        ignore_fields=(
+            # Following fields fails for dcim.rack, see `./source.py` `_FORCE_FIELDS`
+            "created",
+            "last_updated",
+        ),
+        ignore_content_types={
+            "contenttypes.contenttype": None,  # Nautobot has own content types
+            "sessions.session": None,  # Nautobot has own sessions
+            "dcim.cablepath": None,  # Recreated in Nautobot
+            "admin.logentry": None,  # NetBox 3.0 TBD verify
+            "users.userconfig": None,  # NetBox 3.0 TBD verify
+            "auth.permission": None,  # NetBox 3.0 TBD verify
+            "dcim.sitegroup": None,  # NetBox 3.1 TBD verify
+            "ipam.asn": None,  # NetBox 3.1 TBD verify
+            "ipam.iprange": None,  # NetBox 3.1 TBD verify
+            "tenancy.contactgroup": None,  # NetBox 3.1 TBD verify
+            "tenancy.contactrole": None,  # NetBox 3.1 TBD verify
+            "tenancy.contact": None,  # NetBox 3.1 TBD verify
```

Comment:
as this is coming soon to Nautobot, how would we accommodate different options for Nautobot?

Reply:
Will implement inferred skipping of models, not present in Nautobot, so this configuration will go away.